File: tyama_icpc2007dA.R
#!/usr/bin/env Rscript
x=scan("stdin")
i=1
repeat{
  j=x[i]
  if(j==0)break
  y=x[(i+1):(i+j)]
  cat((sum(y)-max(y)-min(y))%/%(length(y)-2))
  cat("\n")
  i=i+j+1
}

File: tyama_icpc2007dA.ex
#!/usr/bin/env elixir
defmodule ICPC2007dA do
	def sum(0,ma,mi,s,z) do
		trunc (s-ma-mi)/(z-2)
	end
	def sum(n,ma,mi,s,z) do
		x=elem Integer.parse(IO.gets('')),0
		sum n-1,Enum.max([ma,x]),Enum.min([mi,x]),s+x,z
	end
	def main2(0) do
	end
	def main2(n) do
		IO.puts sum n,0,1000,0,n
		main2 elem Integer.parse(IO.gets('')),0
	end
	def main() do
		main2 elem Integer.parse(IO.gets('')),0
	end
end
ICPC2007dA.main